functions.php

```diff
@@ -249,3 +244,128 @@ function filter_values ($field)
 {
     return !empty($field['value']);
 }
+
+
+/**
+ * Осуществляет валидацию формы
+ * @return array Массив с сообщениями об ошибках
+ */
+function validate_add_lot_form($con)
+{
+    $data = [];
+    $rules = [
+
+        'lot-name' => function() {
+            return check_required_field('lot-name');
+        },
+
+        'message' => function() {
+            return check_required_field('message');
+        },
+
+        'lot-image' =>  function() {
+            return validate_image('lot-image');
+        },
+
+        'lot-rate' =>  function() {
+            return validate_num('lot-rate');
+        },
+
+        'lot-step' =>  function() {
+            return validate_num('lot-step');
+        },
+
+        'lot-date' =>  function() {
+            return validate_date('lot-date');
+        }
+    ];
+
+    foreach ($_POST as $key => $value) {
+        if (isset($rules[$key])) {
+            $rule = $rules[$key];
+            $check = $rule($con);
+            if (empty($check['error'])) {
+                $data[$key]['value'] = $check['value'];
+            } else {
+                $data[$key]['error'] = $check['error'];
+            }
```

Comment:
Попробуй это сократить до
```suggestion
            $data[$key]['value'] = $rule($con);
```
Возможно сработает )